### DISCUSSION TITLE
Video transparent background generated with gps file only

### DISCUSSION THREAD

**Discussion by xiangzhuyuan** [time4tea/gopro-dashboard-overlay#170] [Q&A] [answered]
Hi, frist of all, thanks for this project.
I am tryming to use gpx or fit file to generate video directly. like this command:

```
>  venv/bin/gopro-dashboard.py --use-fit-only --fit  13511125570_ACTIVITY.fit --font Arial --overlay-size 2160x3840  --units-speed kph --layout xml --layout-xml  default-2160x3840.xml  --profile overlay-mac     13511125570_ACTIVITY.fit.mp4

```
output:

```
Starting gopro-dashboard version 0.123.0
ffmpeg version is 6.1.1
Using Python version 3.11.7 (main, Jan 12 2024, 17:36:22) [Clang 14.0.3 (clang-1403.0.22.14.1)]
GPX/FIT file:     2024-01-15T06:09:00+00:00 -> 2024-01-15T06:42:46+00:00
Generating overlay at Dimension(x=2160, y=3840)
Timeseries has 20261 data points
Processing....
Using *user-defined* profile: overlay-mac
FFMPEG Output is in /var/folders/4h/ypz6kc2j4p1_9xswrwzd3f7ck3tvwt/T/tmpc4kwr09b.txt
Timelapse Factor = 1.000
Layout -> Include component 'date_and_time' = True
Layout -> Include component 'gps_info' = True
Layout -> Include component 'gps-lock' = True
Layout -> Include component 'big_mph' = True
Layout -> Include component 'gradient_chart' = True
Layout -> Include component 'gradient' = True
Layout -> Include component 'altitude' = True
Layout -> Include component 'temperature' = True
Layout -> Include component 'cadence' = True
Layout -> Include component 'heartbeat' = True
Layout -> Include component 'moving_map' = True
Layout -> Include component 'journey_map' = True
Executing ['ffmpeg', '-hide_banner', '-y', '-hide_banner', '-loglevel', 'info', '-f', 'rawvideo', '-framerate', '10.0', '-s', '2160x3840', '-pix_fmt', 'rgba', '-i', '-', '-r', '30', '-vcodec', 'h264_videotoolbox', '-q:v', '65', '13511125570_ACTIVITY.fit.mp4']
```

after got geneated video, and I tried to use `iMove` to make a video, I found the video background somehow it was black, it wasn't transpant.  

Any ideas? thanks.

**Comment by time4tea** [comment, marked as answer]
Hi - Thanks for your interest in the software! - You cannot create a transparent movie in MP4 format, as the file format doesn't support it. You'll need to use a MOV file. There are some examples in the docs https://github.com/time4tea/gopro-dashboard-overlay/tree/main/docs/bin#creating-videos-using-only-gpx-file

**Reply by xiangzhuyuan** [reply]
@time4tea  hi thanks for quick update. I am also checking such knowledge now, don't have enough infos arround video, or ffmpeg.
can i assume that the 1st way in above link, to generate a `mov` file. that files BG is transparent? or I just need a editor could support mov file? thanks

**Reply by time4tea** [reply]
Yes, the default is that the background is transparent. When rendering to a movie format that supports alpha channel (aka transparency) then the background will be transparent, and it should all "just work"...

If you are using a recent version (I can tell that you are!) - it should be just like this:

- did you mean 3840x2160 -> 4k? -> the way you have the size, it will be in portrait mode.
- note that you don't need to specify `--overlay xml` and `--overlay-xml xxx.xml` as you are using the default layout anyway, if you want 4k 
- `--overlay xml` is not required if you use `--overlay-xml`

This should get you going - there are probably other things you can do to make the rendering faster, but this is enough to get started...

```
venv/bin/gopro-dashboard.py --use-fit-only --fit  13511125570_ACTIVITY.fit --overlay-size 3840x2160  --font Arial  --units-speed kph --profile mov activity.mov
```

**Reply by xiangzhuyuan** [reply]
thanks, thats true I am making a portrait mode templare based on the default one you made.
I am still struggling with make a transpaent BG video.

**Comment by xiangzhuyuan** [comment]
@jacksgt could you help to take a look? I saw your helpful blog post. and the image https://github.com/time4tea/gopro-dashboard-overlay/blob/main/examples/2022-11-24-gpx-only-overlay.png too

my env is Mac A1. and python 3.11.7
I am using this command:

```
 venv/bin/gopro-dashboard.py --use-gpx-only --gpx /Users/zhuyuan.xiang/Downloads/test.gpx  --font Arial --overlay-size 2160x3840  --units-speed kph --layout-xml ./gopro_overlay/layouts/default-2160x3840.xml  ~/Downloads/test.mov
```
thank you

**Reply by time4tea** [reply]
Hi, you just needed to copy the example in my last reply..  you are missing `--profile mov`

You need this because both mp4 and mov files are just containers... they can contain lots of different types of image content inside. It's not enough to just change the filename.

Hope that helps.

**Reply by xiangzhuyuan** [reply]
thanks, you are right. I checked @jacksgt 's post agian, and use shortcut to make a video succefully.  👍🏿
